src/group/managed_group/mod.rs

```diff
@@ -802,67 +755,60 @@ impl ManagedGroup {
             .get_credential_bundle(credential.signature_key())
             .ok_or(ManagedGroupError::NoMatchingCredentialBundle)?;
 
-        // If a KeyPackageBundle was provided, create an UpdateProposal
-        let mut plaintext_messages = if let Some(key_package_bundle) = key_package_bundle_option {
-            let update_proposal = self.group.create_update_proposal(
-                &self.aad,
-                &credential_bundle,
-                key_package_bundle.key_package().clone(),
-            )?;
-            self.own_kpbs.push(key_package_bundle);
-            vec![update_proposal]
-        } else {
-            vec![]
-        };
-
         // Include pending proposals into Commit
-        let messages_to_commit: Vec<&MlsPlaintext> = self
-            .pending_proposals
-            .iter()
-            .chain(plaintext_messages.iter())
-            .collect();
-
-        // Create Commit over all proposals
-        // TODO #141
-        let (commit, welcome_option, kpb_option) = self.group.create_commit(
-            &self.aad,
-            &credential_bundle,
-            &messages_to_commit,
-            &[],
-            true, /* force_self_update */
-            None,
-        )?;
-
-        // Add the Commit message to the other pending messages
-        plaintext_messages.push(commit);
+        let messages_to_commit: Vec<&MlsPlaintext> = self.pending_proposals.iter().collect();
```

Comment:
Having to collect something here sounds like the APIs are bad. But that's probably nothing to solve in this PR.